### PR TITLE
fix for issue 1223 (fetchmail persistence idfile)

### LINKED ISSUE #1223: Fetchmail: /var/lib/fetchmail needs persistence
According [fetchmail documentation](https://www.fetchmail.info/fetchmail-man.html#12), an `.idfile` is used to keep track of previously downloaded messages. Shouldn't that file persistent over container restarts?

I'm not a Fetchmail user, perhaps somebody can shine a light on how this currently works?

cc: @Nebukadneza, @hoellen, @kaiyou

### PR DESCRIPTION
**Pull request by Diman0**
## What type of PR?

bug-fix

## What does this PR do?
It introduces a new data folder (/mailu/fetchmail) that will hold the idfile. The file that is used by fetchmail to keep track of what messages where retrieved. Recreating the fetchmail container does not result in all messages being retrieved again. It also configurs fetchmail to actually create this file (--uidl).

It changes fetchmail to run as root. For now this is required, because the mailu data folder (/mailu) is owned by root. In the future we must change all images at the same time, to run without root and use a mailu folder that is not owned by root. That is out of scope for this PR. 

### Related issue(s)
- closes #1223

## Prerequisites
- [x] In case of feature or enhancement: documentation updated accordingly
- [x] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

### CODE REVIEW

**Comment by Diman0** [comment]
@micw This PR introduces a bind mount (/$(data/fetchmail:/data) for fetchmail. Assuming this PR is merged, this means you must reflect the same changes in the helm-charts.

**Comment by nextgens** [comment]
Why ```{{ root }}/fetchmail``` and not ```{{ root }}/data/fetchmail``` (à la #1224) ?

**Comment by Diman0** [comment]
In the current structure we have unique folders for each type of component. 
But using `{{ root }}/data/fetchmail` is also fine with me. It might be a bit confusing, we will have
admin: `{{ root }}/data/*.*`
fetchmail ``{{ root }}/data/fetchmail/*.*``
However that is what that RFC is all about. Move all files according to a well thought out structure.

**Comment by Diman0** [comment]
bors try

**Comment by Diman0** [comment]
If you want to test the PR, then don't forget to add the bind mount to your docker-compose.yml. For example:
```
  fetchmail:
    image: ${DOCKER_ORG:-mailu}/${DOCKER_PREFIX:-}fetchmail:${MAILU_VERSION:-master}
    restart: always
    env_file: mailu.env
    volumes:
      - /mailu/data/fetchmail:/data
    depends_on:
      - resolver
    dns:
      - 192.168.203.254
```

File: optional/fetchmail/fetchmail.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
 FETCHMAIL = """
 fetchmail -N \
+    --idfile /data/.fetchids --uidl \
```
Comment (nextgens) [inline optional/fetchmail/fetchmail.py:16]: I wouldn't use a hidden file here... why prefixing it with a dot? Some users may remove the folder thinking it's empty

Comment (Diman0) [reply optional/fetchmail/fetchmail.py:16]: That is the suggested filename from the fetchmail documentation. I just followed their default file name. It indeed makes more sense to make the file not hidden (removing the leading dot).

**Review by nextgens — APPROVED** [review]
LGTM

**Pull request merged**